tests/functional/Agent.php

```diff
@@ -500,5 +500,112 @@ public function testStaleActions()
+        $this->integer($item->fields['is_deleted'])->isIdenticalTo(1);
+        $this->integer($item->fields['states_id'])->isIdenticalTo($states_id2);
+
+        //create another new status for test while previous status is empty (0)
+        $states_id3 = $state->add(['name' => 'Stale3']);
+        $this->integer($states_id3)->isGreaterThan(0);
+
+        //set last agent contact far ago
+        $DB->update(
+            \Agent::getTable(),
+            ['last_contact' => date('Y-m-d H:i:s', strtotime('-1 year'))],
+            ['id' => $agent['id']]
+        );
+
+        //define sale agents actions
+        \Config::setConfigurationValues(
+            'inventory',
+            [
+                'stale_agents_delay' => 1,
+                'stale_agents_action' => exportArrayToDB([
+                    \Glpi\Inventory\Conf::STALE_AGENT_ACTION_STATUS,
+                    \Glpi\Inventory\Conf::STALE_AGENT_ACTION_TRASHBIN
+                ]),
+                'previous_stale_agents_status' => json_encode([0]),
+                'stale_agents_status' => $states_id3
+            ]
+        );
+
+        //run crontask
+        $task = new \CronTask();
+        $this->integer(\Agent::cronCleanoldagents($task))->isIdenticalTo(1);
+
+        //check item has been updated
+        $this->boolean($item->getFromDB($item->fields['id']))->isTrue();
+        $this->integer($item->fields['is_deleted'])->isIdenticalTo(1);
+        $this->integer($item->fields['states_id'])->isIdenticalTo($states_id3);
+
+        //create another new status for test while previous status is empty (0)
+        $states_id4 = $state->add(['name' => 'Stale4']);
+        $this->integer($states_id4)->isGreaterThan(0);
+
+        //set last agent contact far ago
+        $DB->update(
+            \Agent::getTable(),
+            ['last_contact' => date('Y-m-d H:i:s', strtotime('-1 year'))],
+            ['id' => $agent['id']]
+        );
+
+        //define sale agents actions
+        \Config::setConfigurationValues(
+            'inventory',
+            [
+                'stale_agents_delay' => 1,
+                'stale_agents_action' => exportArrayToDB([
+                    \Glpi\Inventory\Conf::STALE_AGENT_ACTION_STATUS,
+                    \Glpi\Inventory\Conf::STALE_AGENT_ACTION_TRASHBIN
+                ]),
+                'previous_stale_agents_status' => json_encode([
+                    $states_id,
+                    $states_id2,
+                ]),
+                'stale_agents_status' => $states_id4
+            ]
+        );
+
+        //run crontask
+        $task = new \CronTask();
+        $this->integer(\Agent::cronCleanoldagents($task))->isIdenticalTo(1);
+
+        //check item has been updated
+        $this->boolean($item->getFromDB($item->fields['id']))->isTrue();
+        $this->integer($item->fields['is_deleted'])->isIdenticalTo(1);
+        $this->integer($item->fields['states_id'])->isIdenticalTo($states_id3);
```

Comment:
perhaps a final test to check that an asset with a status of "xxx" is not changed when the out-of-date agent configuration needs to

change the status of an asset with a status of "yyyy".